services/motion/builtin/builtin.go

```diff
@@ -231,52 +233,183 @@ func (ms *builtIn) MoveOnGlobe(
+		backgroundWorkers.Add(1)
+		goutils.ManagedGo(func() {
+			ticker := time.NewTicker(period)
+			defer ticker.Stop()
+			for {
+				select {
+				case <-ctx.Done():
+					errChan <- ctx.Err()
+					return
+				case <-ticker.C:
+					if err := fn(ctx); err != nil {
+						errChan <- err
+					}
+					return
+				}
+			}
+		}, backgroundWorkers.Done)
+	}
+
+	// start goroutine to execute plan
+	backgroundWorkers.Add(1)
+	goutils.ManagedGo(func() {
+		for {
+			select {
+			case <-ctx.Done():
+				errChan <- ctx.Err()
+				return
+			case planBase := <-planChan:
+				if err := ms.executePlan(cancelCtx, planBase.base, planBase.plan); err != nil {
+					errChan <- err
+					return
+				}
+
+				position, _, err := movementSensor.Position(cancelCtx, nil)
+				if err != nil {
+					errChan <- err
+					return
+				}
+
+				ms.logger.Infof("position: %v", position)
+				if spatialmath.GeoPointToPose(position, destination).Point().Norm() <= kinematicsOptions.PlanDeviationThresholdMM {
+					successChan <- true
+					return
+				}
+			}
+		}
+	}, backgroundWorkers.Done)
+
+	// loop to monitor things that poll and replan appropriately
+	for {
+		select {
+		case <-ctx.Done():
+			return false, ctx.Err()
+		case err := <-errChan:
+			fmt.Println("this actually happening")
+			return false, err
+		case <-replanChan:
+			fmt.Println("replanning")
+
+			ms.cancelFn()
+			cancelCtx, cancelFn = context.WithCancel(ctx)
+			ms.cancelFn = cancelFn
+
+			plan, kb, err := ms.planMoveOnGlobe(
+				ctx,
+				componentName,
+				destination,
+				movementSensor,
+				obstacles,
+				kinematicsOptions,
+				extra,
+			)
+			if err != nil {
+				return false, err
+			}
+			planChan <- planAndBase{kb, plan}
+
+			// drain the replanChan
+			for len(replanChan) > 0 {
+				<-replanChan
+			}
+
+			startPolling(cancelCtx, positionPollingPeriod, func(ctx context.Context) error {
+				// TODO: the function that actually monitors position
+				fmt.Println("position poll")
+				return nil
+			})
+			startPolling(cancelCtx, obstaclePollingPeriod, func(ctx context.Context) error {
+				// TODO: the function that actually monitors obstacles
+				fmt.Println("obstacle poll")
+				replanChan <- true
+				return nil
+			})
+		case <-successChan:
+			return true, nil
+		}
+	}
+}
+
+func (ms *builtIn) executePlan(ctx context.Context, kinematicBase kinematicbase.KinematicBase, plan motionplan.Plan) error {
+	waypoints, err := plan.GetFrameSteps(kinematicBase.Name().Name)
 	if err != nil {
-		return false, fmt.Errorf("error making plan for MoveOnMap: %w", err)
+		return err
 	}
 
-	// execute the plan
-	for i := 1; i < len(plan); i++ {
-		ms.logger.Info(plan[i])
-		if err := kb.GoToInputs(ctx, plan[i]); err != nil {
-			return false, err
+	for i := 1; i < len(waypoints); i++ {
+		select {
+		case <-ctx.Done():
+			return nil
+		default:
+			ms.logger.Info(waypoints[i])
+			segment := []referenceframe.Input{
```

Comment:
[q] why not just pass in `waypoints[i]` for `GoToInputs`

Reply:
Yeah this needs to be reworked to deal with ptgs, so still more to be done here